corehq/form_processor/utils/general.py

```diff
@@ -9,3 +12,35 @@ def should_use_sql_backend(domain):
     # uncomment when ready for production
     # return USE_SQL_BACKEND.enabled(domain)
     return False
+
+
+def cache_return_value(fn):
+    """
+    Docorator to save the return value of a method or property.
+    Use this where you can't use quickcache or memoize because you can't
+    guarantee a consistent or unique lookup key e.g. forms that get deprecated
```

Comment:
:+1: just the kind of insight I knew you would have. Thanks for helping out my sleep deprived brain!